stk/molecular/molecules/molecule.py

```
@@ -1050,6 +1051,72 @@ def update_from_mol(self, path, conformer=-1):
         self.mol.RemoveConformer(conformer)
         self.mol.AddConformer(conf)
 
+    def update_from_xyz(self, path, conformer=-1):
+        """
+        Updates molecular structure to match a ``.xyz`` file.
+
+        Parameters
+        ----------
+        path : :class:`str`
+            The full path of the ``.mol`` file from which the structure
+            should be updated.
+
+        conformer : :class:`int`, optional
+            The conformer to be updated.
+
+        Returns
+        -------
+        None : :class:`NoneType`
+
+        Raises
+        ------
+        :class:`RuntimeError`
+            If the number of atoms in the file does not match the
+            number of atoms in the molecule or if atom elements in the
+            file do not agree with the atom elements in the molecule.
+
+        """
+
+        if conformer == -1:
+            conformer = self.mol.GetConformer(conformer).GetId()
+
+        with open(path, 'r') as f:
+            atom_count, _, *content = f.readlines()
+
+        # Check the atom count is correct.
+        num_atoms = self.mol.GetNumAtoms()
+        if int(atom_count) != num_atoms:
+            raise RuntimeError(
+                'The number of atoms in the xyz file does '
+                'not match the number of atoms in the molecule.'
+            )
+
+        # Save all the coords in the file.
+        new_coords = []
+        for i, line in enumerate(content):
+            element, *coords = line.split()
+            if element.isnumeric():
+                element = periodic_table[int(element)]
+
```

should raise an error if this statement is False